modulemd/modulemd-modulemetadata.c

```diff
@@ -435,11 +409,12 @@ modulemd_modulemetadata_set_buildrequires (ModulemdModuleMetadata *self,
                                            GHashTable             *buildrequires)
 {
   g_return_if_fail (MODULEMD_IS_MODULEMETADATA (self));
+  g_return_if_fail (buildrequires);
 
-  if (buildrequires != self->buildrequires)
+  if (self->buildrequires != buildrequires)
     {
-      g_hash_table_destroy (self->buildrequires);
-      modulemd_copy_htable_strings (buildrequires, &self->buildrequires);
+      g_hash_table_unref (self->buildrequires);
+      self->buildrequires = g_hash_table_ref (buildrequires);
       g_object_notify_by_pspec (G_OBJECT (self),
```

Comment:
This significantly changes the desired behavior. Please justify it.
The reason for forcing a deep copy here was to ensure that the caller cannot modify the contents of the hash table without triggering a signal (which would happen when saving the property back).

By just bumping the refcount and returning the original value, the contents can be `insert()`ed or `remove()`ed without causing the signal to fire, so consumers could get out of sync.

Furthermore, the commit message is completely useless. The phrase "totally busted" does not explain in what way it didn't work properly or why these changes would improve the situation.

Patch is rejected.